modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java

```diff
@@ -1928,24 +1931,24 @@ private Configuration mergeExistingResourcePluginConfiguration(Resource resource
 
         // If there is no update necessary just return the current plugin config of the existing resource
         Configuration result = resource.getPluginConfiguration();
-
         ConfigurationDefinition configDef = resource.getResourceType().getPluginConfigurationDefinition();
         if (null == configDef) {
             return result;
         }
-
+        Map<String, PropertyDefinition> propDefs = configDef.getPropertyDefinitions();
         Configuration existingPluginConfig = resource.getPluginConfiguration().deepCopy(false);
-        Configuration defaultPluginConfig = ConfigurationUtility.createDefaultConfiguration(configDef);
+
         boolean configChanged = false;
 
         // for each property, update the existing plugin config if discovery has set a non-default value
         for (String propertyName : pluginConfig.getAllProperties().keySet()) {
             Property discoveredProp = pluginConfig.get(propertyName);
-            Property defaultProp = defaultPluginConfig.get(propertyName);
-            if (!discoveredProp.equals(defaultProp)) {
+            // Only overrides the property if it is a read-only property
+            // See BZ: 1379834
+            if (propDefs.get(propertyName).isReadOnly()) {
```

Comment:
the purpose of `!discoveredProp.equals(defaultProp)` Is supposed to check if the newly discovered property is different from the default, then, if it is different updates with the new value, otherwise, it doesn't update. But in most of the cases I saw, the default property is null, (that is because the default values are taken from defaultValue XML attribute) so the condition is always true. But we can still check for that , just in case.